### PR TITLE
New SpecFilter Version 3.0

### PR DESCRIPTION
**Pull request by ralphdoe**
Classes, Methods, Variables Renaming. 
New OpenApiSpecFitler Interface
AbsctractFilter implementation
Starting the new SpecFilter Implementation

### CODE REVIEW

File: modules/swagger-core/src/main/java/io/swagger/core/filter/AbstractSpecFilter.java [hunk 1]
```diff
@@ -1,44 +1,40 @@
...
 
-public abstract class AbstractSpecFilter implements SwaggerSpecFilter {
-    public boolean isOperationAllowed(
-            Operation operation,
-            ApiDescription api,
-            Map<String, List<String>> params,
-            Map<String, String> cookies,
-            Map<String, List<String>> headers) {
-        return true;
+public abstract class AbstractSpecFilter implements OpenAPISpecFilter {
```
Comment (frantuma) [inline modules/swagger-core/src/main/java/io/swagger/core/filter/AbstractSpecFilter.java]: should add a "removeBrokenReferences" to interface and abstract class, similar to 
https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-core/src/main/java/io/swagger/core/filter/AbstractSpecFilter.java#L42

File: modules/swagger-core/src/test/java/io/swagger/filter/InternalModelPropertiesRemoverFilter.java [hunk 2]
```diff
@@ -11,21 +11,5 @@
  * "super-user" is passed
  */
 public class InternalModelPropertiesRemoverFilter extends AbstractSpecFilter {
```
Comment (frantuma) [inline modules/swagger-core/src/test/java/io/swagger/filter/InternalModelPropertiesRemoverFilter.java]: should implement method similar to previous version, and related tests

File: modules/swagger-core/src/test/java/io/swagger/filter/NoGetOperationsFilter.java [hunk 3]
```diff
@@ -1,26 +1,9 @@
...
-import io.swagger.model.ApiDescription;
-import io.swagger.oas.models.Operation;
-
-import java.util.List;
-import java.util.Map;
 
 /**
  * Sample filter to avoid all get operations for the resource
  **/
 public class NoGetOperationsFilter extends AbstractSpecFilter {
```
Comment (frantuma) [inline modules/swagger-core/src/test/java/io/swagger/filter/NoGetOperationsFilter.java]: same comment as above

File: modules/swagger-core/src/test/java/io/swagger/filter/NoUserOperationsFilter.java [hunk 4]
```diff
@@ -1,26 +1,10 @@
...
-            ApiDescription api,
-            Map<String, List<String>> params,
-            Map<String, String> cookies,
-            Map<String, List<String>> headers) {
-        if (api.getPath().startsWith("/user")) {
-            return false;
-        }
-        return true;
-    }
+
```
Comment (frantuma) [inline modules/swagger-core/src/test/java/io/swagger/filter/NoUserOperationsFilter.java]: same comment as above

File: modules/swagger-core/src/test/java/io/swagger/filter/RemoveInternalParamsFilter.java [hunk 5]
```diff
@@ -13,23 +13,5 @@
...
-            if (headers != null) {
-                if (headers.containsKey("super-user")) {
-                    return true;
-                }
-            }
-            return false;
-        }
-        return true;
-    }
+
```
Comment (frantuma) [inline modules/swagger-core/src/test/java/io/swagger/filter/RemoveInternalParamsFilter.java]: same comment as above

File: modules/swagger-core/src/test/java/io/swagger/filter/SpecFilterTest.java [hunk 6]
```diff
@@ -318,14 +318,14 @@ public void filterWithNullDefinitions() throws IOException {
...
             result.add(item.getName());
         }
         return result;
     }
 
-    private OpenAPI getSwagger(String path) throws IOException {
+    private OpenAPI getOpenAPI(String path) throws IOException {
         final String json = ResourceUtils.loadClassResource(getClass(), path);
         return Json.mapper().readValue(json, OpenAPI.class);
     }
```
Comment (frantuma) [inline modules/swagger-core/src/test/java/io/swagger/filter/SpecFilterTest.java]: should enable and implement/fix tests

File: modules/swagger-core/src/main/java/io/swagger/core/filter/SpecFilter.java [hunk 7]
```diff
@@ -4,315 +4,61 @@
...
-            return new HashSet<String>(Arrays.asList(((RefModel) model).getSimpleRef()));
-        }
-        */
-        if(model.get$ref() != null) {
-            return new HashSet<String>(Arrays.asList((model.get$ref())));
-        }
-        return null;
+        return operation;
     }
 }
```
Comment (frantuma) [inline modules/swagger-core/src/main/java/io/swagger/core/filter/SpecFilter.java]: implement execution of other filter ops param, property, etc..)

**Comment by arthurdm** [comment]
@ralphdoe @frantuma - I would suggest we change the package of OpenApiSpecFitler to io.swagger.oas.core, to match the annotations and models packages.

**Comment by frantuma** [comment]
replaced by #2423

**Pull request closed without merging**